prediction_market_agent/agents/crewai_subsequential_agent/benchmark.py

```diff
@@ -0,0 +1,159 @@
+from prediction_market_agent_tooling.benchmark.utils import (
+    OutcomePrediction,
+    Prediction,
+)
+from prediction_market_agent_tooling.gtypes import Probability, DatetimeWithTimezone
+from prediction_market_agent_tooling.markets.markets import AgentMarket
+from prediction_market_agent_tooling.tools.utils import utcnow
+from pydantic import BaseModel
+
+from prediction_market_agent.agents.crewai_subsequential_agent.crewai_agent_subquestions import CrewAIAgentSubquestions
+from prediction_market_agent.agents.known_outcome_agent.known_outcome_agent import (
+    Result,
+    get_known_outcome,
+)
+
+
+def build_market_from_question_without_validation(question: str) -> Market:
+    return Market(url=question,
+                  question=question, p_yes=0.5,
+                  source=MarketSource.MANIFOLD,
+                  volume=0,
+                  created_time=DatetimeWithTimezone(datetime(2024, 1, 1)),
+                  close_time=DatetimeWithTimezone(datetime(2024, 3, 15))
+                  )
+
+
+def build_binary_agent_market_from_question(question: str) -> AgentMarket:
+    return AgentMarket(
+        id=question,
+        question=question,
+        p_yes=Probability(0.5),
+        created_time=datetime(2024, 1, 1),
+        resolution=None,
+        outcomes=["YES", "NO"],
+    )
+
+
+class QuestionAndAnswer(BaseModel):
+    question: str
+    result: Result
+    bet_correct: bool
+
+
+class CrewAIAgentSubquestionsBenchmark(AbstractBenchmarkedAgent):
+    def __init__(
+            self,
+            agent_name: str,
+            max_workers: int,
+            model: str,
+            max_tries: int,
+    ) -> None:
+        self.model = model
+        self.max_tries = max_tries
+        self.agent = CrewAIAgentSubquestions()
+        super().__init__(agent_name=agent_name, max_workers=max_workers)
+
+    def predict(self, market_question: str) -> Prediction:
+
+        market = build_binary_agent_market_from_question(market_question)
+        result = self.agent.answer_binary_market(market)
+
+        answer = get_known_outcome(
+            model=self.model,
+            question=market_question,
+            max_tries=self.max_tries,
+        )
+        print(f"Answered {market_question=} with {answer.result=}, {answer.reasoning=}")
+        if not answer.has_known_result():
+            return Prediction(
+                is_predictable=False,
+                outcome_prediction=None,
+            )
+        else:
+            return Prediction(
+                is_predictable=True,
+                outcome_prediction=OutcomePrediction(
+                    p_yes=answer.result.to_p_yes(),
+                    confidence=1.0,
+                    info_utility=None,
+                ),
+            )
+
+
+if __name__ == "__main__":
+    load_dotenv()
+    tomorrow_str = (utcnow() + timedelta(days=1)).strftime("%d %B %Y")
+
+    # Fetch example questions which our agents answered in the past.
+    questions = [
```

Comment:
@evangriffiths Correct me if I am wrong please, but I think you used these hard-coded questions, because you indeed needed to validate your experiment on already known markets, right?

But this agent should be useful generally for future predictions.

So here, questions shouldn't be hard-coded, but rather fetched as in EVO: https://github.com/polywrap/evo.researcher/blob/9225e842c34bc6eb03703c08e61a9947ee307379/scripts/benchmark.py

Reply:
Additionally, what do you guys think about doing this benchmark in two ways, as in EVO:

1. On not-yet-resolved markets, so the benchmark will compare agent's probability with manifold's probability.
2. On already resolved markets, but using the time restriction approach. 

I'm not sure about how easy it will be to use (2) here, you can check how it's implemented in EVO agents: https://github.com/polywrap/evo.researcher/blob/7c89df712df21cb5e7394509831473546436cc30/prediction_prophet/benchmark/agents.py#L196 (the idea there is to gather links from internet, but before using them, filter them out using `url_is_older_than` function)

Reply:
Yes, that's right. Sorry @gabrielfior, for not being more clear but in this case it makes more sense to just use `get_binary_markets` instead of hardcoded markets. Can start with that and look at (2) as peter suggests after I reckon

Reply:
Fixed.